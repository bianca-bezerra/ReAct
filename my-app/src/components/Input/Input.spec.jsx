import { render, screen } from '@testing-library/react'
import { Input } from '.'
import userEvent from '@testing-library/user-event'

describe('<Input/>', () => {
  it('should have a searchInput', () => {
    const fn = jest.fn()
    render(<Input handleChange={fn} searchInput={'teste'} />)
    const input = screen.getByPlaceholderText("Search...")

    expect(input).toBeInTheDocument()
    expect(input.value).toBe('teste')
  })

  it('should call handleChange on each key pressed', () => {
    const fn = jest.fn()
    render(<Input handleChange={fn} />)
    const input = screen.getByPlaceholderText("Search...")
    const value = 'valor do input'
    userEvent.type(input, value)
    expect(input.value).toBe(value)
    expect(fn).toHaveBeenCalledTimes(value.length)
  })

  it('should match snapshot', () => {
    const fn = jest.fn()
    const { container } = render(<Input handleChange={fn} searchInput={'teste'} />)
    expect(container.firstChild).toMatchSnapshot()
  })

})
