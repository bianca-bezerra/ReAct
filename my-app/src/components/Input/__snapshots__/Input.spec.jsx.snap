// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> should match snapshot 1`] = `
<input
  class="search-input"
  placeholder="Search..."
  type="text"
  value="teste"
/>
`;
